objectherkenning_openbare_ruimte/on_edge/detection_pipeline/components/model_result.py

```diff
@@ -0,0 +1,314 @@
+        for key, value in self.result.speed.items():
+            speed_str = speed_str + f"{key}: {value:.2f}ms, "
+        speed_str = speed_str[0:-2] + "}"
+
+        return [obj_str, speed_str]
+
+    def yolo_annotation_to_bounds(
+        self, yolo_annotation: str, img_shape: Tuple[int, int]
+    ) -> Tuple[int, int, int, int]:
+        """
+        Convert YOLO annotation with normalized values to absolute bounds.
+
+        Parameters
+        ----------
+        yolo_annotation : str
+            YOLO annotation string in the format:
+            "<class_id> <x_center_norm> <y_center_norm> <w_norm> <h_norm>".
+        img_shape : Tuple[int, int]
+            Image dimensions as tuple (width, height)
+
+        Returns
+        -------
+        tuple
+            A tuple (x_min, y_min, x_max, y_max).
+        """
+        _, x_center_norm, y_center_norm, w_norm, h_norm = map(
+            float, yolo_annotation.split()[0:5]
+        )
+
+        x_center_abs = x_center_norm * img_shape[0]
+        y_center_abs = y_center_norm * img_shape[1]
+        w_abs = w_norm * img_shape[0]
+        h_abs = h_norm * img_shape[1]
+
+        x_min = int(x_center_abs - (w_abs / 2))
+        y_min = int(y_center_abs - (h_abs / 2))
+        x_max = int(x_center_abs + (w_abs / 2))
+        y_max = int(y_center_abs + (h_abs / 2))
+
+        return x_min, y_min, x_max, y_max
+
+    def blur_inside_boxes(
+        self,
+        boxes: Union[List[Tuple[float, float, float, float]], npt.NDArray[np.float_]],
+        blur_kernel_size: int = 165,
+        box_padding: int = 0,
+    ):
+        """
+        Apply GaussianBlur with given kernel size to the area given by the bounding box(es).
+
+        Parameters
+        ----------
+        boxes : List[Tuple[float, float, float, float]]
+            Bounding box(es) of the area(s) to blur, in the format (xmin, ymin, xmax, ymax).
+        blur_kernel_size : int (default: 165)
+            Kernel size (used for both width and height) for GaussianBlur.
+        box_padding : int (default: 0)
+            Optional: increase box by this amount of pixels before applying the blur.
+
+        Return
+        ------
+        image : numpy.ndarray
+            The image blurred.
+        """
+        img_height, img_width, _ = self.image.shape
+
+        for box in boxes:
+            x_min, y_min, x_max, y_max = map(int, box)
+
+            x_min = max(0, x_min - box_padding)
+            y_min = max(0, y_min - box_padding)
+            x_max = min(img_width, x_max + box_padding)
+            y_max = min(img_height, y_max + box_padding)
+
+            logger.debug(f"Blurring inside: {(x_min, y_min)} -> {(x_max, y_max)}")
+            area_to_blur = self.image[y_min:y_max, x_min:x_max]
+            blurred = cv2.GaussianBlur(
+                area_to_blur, (blur_kernel_size, blur_kernel_size), 0
+            )
+            self.image[y_min:y_max, x_min:x_max] = blurred
+
+    def blur_outside_boxes(
```

Comment:
Just a note: `blur_outside_boxes` and `crop_outside_boxes` are currently not used, but these are part of the different scenarios that are investigated by the citizen panel. So we may need them later.